src/instructions/port.rs

```diff
@@ -66,43 +67,43 @@ impl PortWrite for u32 {
     }
 }
 
-mod sealed {
-    pub trait Access {
-        const DEBUG_NAME: &'static str;
-    }
-}
+/// A marker trait for access types which allow accessing port values.
+pub trait PortAccess: Sealed {}
```

Comment:
Do we need this `PortAccess` trait?

It only seems to be used to implement `PortGeneric: Debug`, but we could just use `Sealed` for that.

Reply:
My initial reasoning for this was to restrict the `Debug` impl to only be available with the port access types like before i.e. it would be implemented only for `PortGeneric<T, ReadWriteAccess>`, `PortGeneric<T, ReadOnlyAccess>` and `PortGeneric<T, WriteOnlyAccess>`. If we only require the access type to implement `Sealed`, we would also implement `Debug` for other types such as `PortGeneric<T, Size4KiB>`.

Ultimately the trade-off  here is being slightly more correct at the cost of introducing a seemingly useless trait. I'd also be fine with relaxing the bounds to implement `Debug` slightly to simplify.

Reply:
Seems reasonable